webservices/load_legal_docs.py

```diff
@@ -247,3 +261,182 @@ def load_advisory_opinions_into_s3():
                 print("%d of %d advisory opinions written to s3" % (i + 1, len(new_docs)))
         else:
             print("No new advisory opinions found.")
+
+def process_mur_pdf(mur_no, pdf_key, bucket):
+    response = requests.get('http://www.fec.gov/disclosure_data/mur/%s.pdf'
+                            % mur_no, stream=True)
+
+    with NamedTemporaryFile('wb+') as pdf:
+        for chunk in response:
+            pdf.write(chunk)
+
+        pdf.seek(0)
+        try:
+            pdf_doc = slate.PDF(pdf)
+        except:
+            print('Could not parse pdf: %s' % pdf_key)
+            pdf_doc = []
+        pdf_pages = len(pdf_doc)
+        pdf_text = ' '.join(pdf_doc)
+        pdf_size = getsize(pdf.name)
+
+        pdf.seek(0)
+        bucket.put_object(Key=pdf_key, Body=pdf,
+                          ContentType='application/pdf', ACL='public-read')
+        return pdf_text, pdf_size, pdf_pages
+
+def get_subject_tree(html, tree=[]):
+    # get next token
+    root = re.match("([^<]+)(?:<br>)?(.*)", html, re.S)
+    list_item = re.match("<li>(.*?)</li>(.*)", html, re.S)
+    unordered_list = re.match("<ul\s+class='no-top-margin'>(.*)", html, re.S)
+    end_list = re.match("</ul>(.*)", html, re.S)
+    empty = re.match("\s*<br>\s*", html)
+
+    if empty:
+        pass
+    elif end_list:
+        # reduce
+        sub_tree = []
+        node = tree.pop()
+        while node != 'ul_start':
+            sub_tree.append(node)
+            node = tree.pop()
+
+        sub_tree.reverse()
+        if tree[-1] != 'ul_start':
+            tree[-1]['children'] = sub_tree
+        else:
+            tree.extend(sub_tree)
+    elif unordered_list:
+        # shift
+        tree.append('ul_start')
+    elif list_item or root:
+        # shift
+        subject = re.sub('\s+', ' ', (list_item or root).group(1))\
+                    .strip().lower().capitalize()
+        tree.append({'text': subject})
+    else:
+        print(html)
+        raise Exception("Could not parse next token.")
+
+    if not empty:
+        tail = (root or list_item or unordered_list or end_list)
+        if tail:
+            html = tail.groups()[-1].strip()
+            if html:
+                get_subject_tree(html, tree)
+    return tree
+
+def get_citations(data):
+    citation_texts = re.findall("(.*?)<br>", data)
+
+    us_codes = []
+    regulations = []
+
+    for citation_text in citation_texts:
+        us_code_match = re.match("([0-9]+) U\.S\.C\. ([0-9]+)", citation_text)
+        regulation_match = re.match("([0-9]+) C\.F\.R\. ([0-9]+)(?:\.([0-9]+))?", citation_text)
+
+        if us_code_match:
+            url = 'http://api.fdsys.gov/link?collection=uscode&' +\
+                  'title=%s&year=mostrecent&section=%s'\
+                  % (us_code_match.group(1), us_code_match.group(2))
```

Comment:
This might be a bit nit-picky, but to be on the defensive side we might want to url encode these parameters. The regex would prevent anything fishy, but just incase of future refactoring. We don't have an explicit test case around this.